src/uiElements/components/select/selectOption.js

```diff
@@ -46,35 +45,79 @@ var DropDownOptions = React.createClass({
 	},
 
 	renderClassName: function() {
-		return classNames('select-option', this.props.className);
+		if (this.props.className) {
+			if (this.props.options.length >= 6) {
+				return "select-option-search " + this.props.className
+			} else {
+				return "select-option " + this.props.className
+			}
+		} else {
+			if (this.props.options.length >= 6) {
+				return "select-option-search"
+			} else {
+				return "select-option"
+			}
+		}
  	},
 
- 	onButtonItemClicked: function(value) {
+ 	renderSearchBar: function() {
+ 		var length = this.props.options.length;
+
+ 		if (length >= 6) {
+ 			if (this.state.isDisplayingList) {
+	 			return (
+	 				<div className="select-option-search-bar">
+	 					<SearchBar isSmall={true} onChange={this.onSearchChanged}/>
+	 				</div> 
+	 			)
+	 		}
+ 		}
+ 	},
+
+ 	onSearchChanged: function(text) {
+ 		
+ 		this.props.onSearchChange(text);
+ 	},
+
+ 	onButtonItemClicked: function(item) {
  		this.setState({
  			isDisplayingList: false,
-			value: value
+			value: item.text
```

Comment:
the state name should probably change.` this.state.value` is what gets displayed to the user on the button.